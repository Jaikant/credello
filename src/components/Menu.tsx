import * as React from 'react';
import styled from 'styled-components';
import { slide as BurgerMenu } from 'react-burger-menu';
import MenuIcon from '@material-ui/icons/Menu';
import CloseIcon from '@material-ui/icons/Close';
import { useMediaQuery } from 'react-responsive';

const MenuOpenIcon = styled(MenuIcon)`
  color: ${({ theme }) => theme.colors.grey};
`;

const MenuCloseIcon = styled(CloseIcon)`
  color: ${({ theme }) => theme.colors.grey};
`;

const menuStyles = {
  bmBurgerButton: {
    position: 'relative',
    width: '36px',
    height: '30px',
    left: '16px',
    top: '10px',
  },
};

const StyledBurger = styled(BurgerMenu)`
  width: 50%;
  .bm-item {
    text-align: center;
    display: inline-block;
    text-decoration: none;
    margin-bottom: 5vh;
    color: #d1d1d1;
    transition: color 0.2s;
  }
  .bm-item:hover {
    color: white;
  }
  .bm-menu-wrap {
    position: fixed;
    height: 100%;
    width: 50%;
  }
  ,
  .bm-burger-bars {
    background: #373a47;
  }
  .bm-burger-bars-hover: {
    background: #a90000;
  }
  ,
  .bm-cross-button {
    height: 24px;
    width: 24px;
  }
  .bm-cross {
    background: #fff;
  }
  .bm-menu {
    background: #fff;
    padding: 3em 1em;
    font-size: 1.2em;
  }
  .bm-morph-shape {
  }
  .bm-item-list {
  }
  .bm-overlay {
    background: rgba(0, 0, 0, 0.3);
  }
`;

export default () => {
  const isDesktopOrLaptop = useMediaQuery({ minDeviceWidth: 1224 });
  const [menuOpen, setMenuOpen] = React.useState(false);
  const [width, setWidth] = React.useState('80%');

  React.useEffect(() => {
    if (isDesktopOrLaptop) {
      setWidth('25%');
    }
  }, []);

  const showSettings = (event) => {
    event.preventDefault();
  };

  const isMenuOpen = (state) => {
    if (state.isOpen != menuOpen) {
      setMenuOpen(state.isOpen);
    }
  };

  return (
    <StyledBurger
      customBurgerIcon={<MenuOpenIcon />}
      customCrossIcon={<MenuCloseIcon />}
      width={width}
      isOpen={menuOpen}
      onStateChange={isMenuOpen}
      pageWrapId={'page-wrap'}
      outerContainerId={'outer-container'}
      styles={menuStyles}
    >
      <a id="home" className="menu-item" href="/">
        Home
      </a>
      <a id="about" className="menu-item" href="/about">
        About
      </a>
      <a id="contact" className="menu-item" href="/contact">
        Contact
      </a>
      <a onClick={showSettings} className="menu-item--small" href="">
        Settings
      </a>
    </StyledBurger>
  );
};
