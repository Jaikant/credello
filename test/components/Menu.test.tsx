import React from 'react';
import { render } from '../testUtils';
import Menu from '../../src/components/Menu';
import 'jest-styled-components';

describe('Home page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Menu />, {});
    expect(asFragment()).toMatchSnapshot();
  });
});
