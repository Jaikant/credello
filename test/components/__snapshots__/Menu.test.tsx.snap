// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  .c2 {
  color: #02254d;
}

.c1 {
  color: grey;
}

.c0 {
  width: 50%;
}

.c0 .bm-item {
  text-align: center;
  display: inline-block;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin-bottom: 5vh;
  color: #d1d1d1;
  -webkit-transition: color 0.2s;
  transition: color 0.2s;
}

.c0 .bm-item:hover {
  color: white;
}

.c0 .bm-menu-wrap {
  position: fixed;
  height: 100%;
  width: 50%;
}

.c0,
.c0 .bm-burger-bars {
  background: #373a47;
}

.c0 .bm-burger-bars-hover: {
  background: #a90000;
}

.c0,
.c0 .bm-cross-button {
  height: 24px;
  width: 24px;
}

.c0 .bm-cross {
  background: #fff;
}

.c0 .bm-menu {
  background: #fff;
  padding: 3em 1em;
  font-size: 1.2em;
}

.c0 .bm-overlay {
  background: rgba(0,0,0,0.3);
}

<div>
    <div
      class="bm-overlay"
      style="position: fixed; z-index: 1000; width: 100%; height: 100%; background: rgba(0, 0, 0, 0.3); opacity: 0; transform: translate3d(100%, 0, 0); transition: opacity 0.3s, transform 0s 0.3s;"
    />
    <div
      class="bm-menu-wrap c0"
      id=""
      style="position: fixed; z-index: 1100; width: 80%; height: 100%; transform: translate3d(-100%, 0, 0); transition: all 0.5s;"
    >
      <div
        class="bm-menu"
        style="height: 100%; box-sizing: border-box; overflow: auto;"
      >
        <nav
          class="bm-item-list"
          style="height: 100%;"
        >
          <a
            class="bm-item menu-item"
            href="/"
            id="home"
            style="display: block;"
            tabindex="-1"
          >
            Home
          </a>
          <a
            class="bm-item menu-item"
            href="/about"
            id="about"
            style="display: block;"
            tabindex="-1"
          >
            About
          </a>
          <a
            class="bm-item menu-item"
            href="/contact"
            id="contact"
            style="display: block;"
            tabindex="-1"
          >
            Contact
          </a>
          <a
            class="bm-item menu-item--small"
            href=""
            style="display: block;"
            tabindex="-1"
          >
            Settings
          </a>
        </nav>
      </div>
      <div>
        <div
          class="bm-cross-button"
          style="position: absolute; width: 24px; height: 24px; right: 8px; top: 8px;"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root c1 bm-cross"
            focusable="false"
            style="width: 100%; height: 100%;"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
          <button
            style="position: absolute; left: 0px; top: 0px; width: 100%; height: 100%; margin: 0px; padding: 0px; font-size: 0px; background: transparent; cursor: pointer;"
            tabindex="-1"
          >
            Close Menu
          </button>
        </div>
      </div>
    </div>
    <div>
      <div
        class="bm-burger-button"
        style="z-index: 1000; position: relative; width: 36px; height: 30px; left: 16px; top: 10px;"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root c2 bm-icon"
          focusable="false"
          style="width: 100%; height: 100%;"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
        <button
          style="position: absolute; left: 0px; top: 0px; width: 100%; height: 100%; margin: 0px; padding: 0px; font-size: 0px; background: transparent; cursor: pointer;"
        >
          Open Menu
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
